tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/ReadGitConfigFile.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Tasks;
+using Microsoft.Build.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ReadGitConfigFile : Task
+    {
+        [Required]
+        public string File { get; set; }
+
+        /// <summary>
+        /// An item for each submodule section that has "Path" and "Url" variables. Includes
+        /// certain variable values as metadata.
+        /// 
+        /// ItemSpec = subsection (submodule name)
+        /// Path = value of the "path" variable
+        /// Url = value of the "url" variable
+        /// Branch = value of the "branch" variable, or empty string
+        /// SourceBuildAutoUpdate = value of the "sourceBuildAutoUpdate" variable, or empty string
+        /// </summary>
+        [Output]
+        public ITaskItem[] SubmoduleConfiguration { get; set; }
+
+        public override bool Execute()
+        {
+            var r = new Exec
+            {
+                BuildEngine = BuildEngine,
+                Command = $"git config --file {File} --list",
+                ConsoleToMSBuild = true
+            };
+            if (!r.Execute())
+            {
+                return false;
+            }
+
+            var options = r.ConsoleOutput
+                .Select(item => ConfigOption.Parse(item.ItemSpec))
+                .ToArray();
+
+            SubmoduleConfiguration = options
+                .Where(o => o.Section == "submodule")
+                .GroupBy(o => o.Subsection)
+                .Select(g =>
+                {
+                    string path = g.FirstOrDefault(o => o.Name == "path")?.Value;
+                    string url = g.FirstOrDefault(o => o.Name == "url")?.Value;
+
+                    if (string.IsNullOrEmpty(path) ||
+                        string.IsNullOrEmpty(url))
+                    {
+                        return null;
```

Comment:
You may want/need to filter these `null` items out.  That way you aren't returning `null` items to MSBuild.  I've never tried this before.

Reply:
Yeah, it was working, but no need to rely on it. Adding a clause.